Add release notes for Metals Radium release (#752)
scalameta/metals: website/blog/2019-06-11-radium.md

---
author: Tomasz Godzik
title: Metals v0.6.1 - Radium
authorURL: https://twitter.com/tomekgodzik
authorImageURL: https://avatars1.githubusercontent.com/u/3807253?s=460&v=4
---

We are excited to announce the release of Metals v0.6.1, codename "Radium" 🎉 The
release mostly focused on easier integration with the most popular build tools
such as Mill, Gradle and Maven.

Metals is a language server for Scala that works with VS Code, Atom, Vim,
Sublime Text and Emacs. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) along with contributors from the
community.

In this release we merged 24 PRs and closed 6 issues, full details:
https://github.com/scalameta/metals/milestone/7?closed=1.

## TL;DR

- auto-import with gradle, maven and mill
- better handling of requests that access the presentation compiler
- code completions on keywords

> - VS code doesn't compile projects until it is focused

Reviewer: I propose we pick a consistent term for vscode and stick to it. I propose "VS Code" (space after VS and capital C)

- Metals is now a default server in the Sublime LSP package

Check out the website and give Metals a try: https://scalameta.org/metals/

## Auto-import with gradle, maven and mill

Just as with `sbt` when you open a new project that contains scala code and is

> built with `Gradle`, `Maven` or `Mill`, a pop-up will show prompting you to

Reviewer: The backticks seem unnecessary for tools names. I would use sbt, Gradle, Maven and Mill without backticks

import the project. Once you click it the auto-import will start. There is more

> information about each of the build tools supported in the documentation.

Reviewer: link to the documentation?

> ## Better handling of requests to the presentation compiler

Reviewer: This entire section seems a bit too focused on implementation details. I would rephrase it from a Metals user perspective and link to the relevant issue for the technical details.

Something like:

> Due to how requests to the presentation compiler worked, the Metals process would fail to exit properly in some cases. We've refactored the communication to make it more resilient, so now Metals should not be hanging around longer than necessary!

> As a bonus, this approach allows Metals to prioritize the most recently submitted request when choosing between competing hover/signatureHelp/completion requests.

Previously, we used .synchronized to limit multi-threaded access to the
presentation compiler. This approach had several issues, of which most
critically we risked blocking threads forever preventing the Metals process from
exiting.

Now, the APIs that access the presentation compiler return CompletableFuture and
the implementation no longer uses .synchronized. This change had a nice
side-effect that Metals now prioritizes the most recently submitted request when
choosing between competing hover/signatureHelp/completion requests.

> ## Code completions on keywords

Reviewer: I would add a minor sentence as introduction here:

> Language keywords are now included in the auto-completions. For instance:

```scala
object Main {
    imp@@
}
```

will automatically give you suggestion to do:

```scala
object Main {
    import
}

> ```

Reviewer: > Keywords are suggested based on the context (e.g. you won't see `throw` suggested if the cursor is not inside a declaration)

> ## VS code doesn't compile projects until it is focused

Reviewer: See comment above about VS Code casing.

When you edit your code outside the editor, like in case of switching git
branches, VSCode will wait until it is focused again to start the compilation

> again. This helps to avoid uneccessary compilations ans strain on the processor.

Reviewer: ```suggestion
again. This helps to avoid unnecessary compilations when you're not actively working on the code.
```
